cfht2caom2/preview_augmentation.py

```diff
@@ -594,26 +600,31 @@ def _do_spirou_intensity_spectrum(self):
         # Polarization scale factor
 
         if self._storage_name.suffix in ['e', 't']:
-            sw2d = spirou['WaveAB'].data  # wavelength array (nm)
-            si2d = spirou['FluxAB'].data  # intensity array (normalized)
+            sw2d = spirou['WaveAB'].data.copy()  # wavelength array (nm)
+            si2d = spirou['FluxAB'].data.copy()  # intensity array (normalized)
             sw = np.ravel(sw2d)
             si = np.ravel(si2d)
 
         if self._storage_name.suffix == 'p':
-            sw2d = spirou['WaveAB'].data  # wavelength array (nm)
-            si2d = spirou['StokesI'].data  # intensity array (normalized)
-            sp2d = spirou['Pol'].data  # Pol Stokes array
+            sw2d = spirou['WaveAB'].data.copy()  # wavelength array (nm)
+            si2d = spirou['StokesI'].data.copy()  # intensity array (normalized)
+            sp2d = spirou['Pol'].data.copy()  # Pol Stokes array
             sw = np.ravel(sw2d)
             si = np.ravel(si2d)
             sp = np.ravel(sp2d)
             pScale = 5.0 * max(si)
 
         if self._storage_name.suffix == 's':
             # using uniform wavelength bins
-            sw = spirou[1].data.field(0)
-            si = spirou[1].data.field(1)
+            sw = spirou[1].data.copy().field(0)
+            si = spirou[1].data.copy().field(1)
 
         spirou.close(self._science_fqn)
+        # astropy says
+        # https://docs.astropy.org/en/stable/io/
+        # fits/index.html#working-with-large-files
+        del spirou[0].data
```

Comment:
You've opened multiple extensions. Maybe the safest would be to try to delete `data` from each of them?